python/ray/data/_internal/progress_bar.py

```diff
@@ -83,6 +87,12 @@ def __init__(
                 needs_warning = False
             self._bar = None
 
+    def _truncate_name(self, name: str) -> str:
+        ctx = ray.data.context.DataContext.get_current()
+        if ctx.enable_progress_bar_name_truncation and len(name) > self.MAX_NAME_LENGTH:
+            return name[: self.MAX_NAME_LENGTH - 3] + "..."
```

Comment:
Nit: (Don't need to do this, just throwing it out as an idea) I'm wondering if we should include some of the text at the end. So, for example, instead of:

```
Map(spam)->Map(ham)->...
```

We could do something like

```
Map(spam)->...->Map(ham)
```

Might make it clearer which operator it ends on.

Reply:
yeah sounds good. now the updated output looks like:

```
✔️  Dataset execution finished in 32.55 seconds: 100%|█████████████████████████████████████████████████████████████████████████| 20/20 [00:32<00:00,  1.63s/ bundle]]
- ReadCSV->SplitBlocks(20): 0 active, 0 queued, [cpu: 0.0, objects: 0.0B]: : 20 bundle [00:32,  1.63s/ bundle]]name): 3 active, 17 queued, [cpu: 3.0, objects: 768.0
- MapBatches(f_with_really_long_name)->MapBatches(f_with_really_long_name)->...->MapBatches(f_with_really_long_name): 0 active, 0 queued, [cpu: 0.0, objects: 840.0B
```